### ISSUE TITLE
ipv6 validation allows invalid ipv6 addresses

### ISSUE THREAD

**Issue by kurtextrem** [fabian-hiller/valibot#206]
The current regexp used for checking ipv6 in valibot might not be correct, as it allows, e.g.:
`fe80::3:bEFf:5b:%3NG` which is not valid.

https://github.com/fabian-hiller/valibot/blob/982045e5597f2906714a9f38674880d8ee649645/library/src/validations/ipv6/ipv6.ts#L13

As per https://github.com/garycourt/uri-js/issues/40#issuecomment-532204167, it looks like other regexps can not be used to fully make sure an ipv6 address is valid and the regexp used in uri-js seems quite expensive (although the issue is from 2019, so it might have changed nowadays).
In any case, the Fastify team is now using a different approach to parse ipv6 addresses: [fastify/fast-uri@main/lib/utils.js#L27](https://github.com/fastify/fast-uri/blob/main/lib/utils.js?rgh-link-date=2023-10-07T17%3A52%3A23Z#L27) (the mentioned "read buffer" approach is used there).

The buffer code is faster, but also increases bundle size a lot more compared to the regex. What do we prefer?

**Comment by fabian-hiller** [comment]
Fastify's code seems to be quite long. What if we use a mix of regex and JS code? The regex checks the basic structure and the JS code checks the details. This is what I found on Medium: https://medium.com/@stheodorejohn/validating-ipv4-and-ipv6-addresses-with-ease-unveiling-the-power-of-validation-in-javascript-2af04ee065c5

**Comment by kurtextrem** [comment]
The Medium approach sounds good to me. Do you plan to make the change?

**Comment by fabian-hiller** [comment]
Yes, I plan to improve that in the long run. However, I will not have the time in the next few weeks for it.

**Comment by fabian-hiller** [comment]
Thank you for the tip. Currently my focus is on other areas of the library but I will get back to this in the long run.
